usermods/Battery/usermod_v2_Battery.h

```diff
@@ -0,0 +1,783 @@
+    int8_t lowPowerIndicatorPreset = USERMOD_BATTERY_LOW_POWER_INDICATOR_PRESET;
+    int8_t lowPowerIndicatorThreshold = USERMOD_BATTERY_LOW_POWER_INDICATOR_THRESHOLD;
+    int8_t lowPowerIndicatorReactivationThreshold = lowPowerIndicatorThreshold+10;
+    int8_t lowPowerIndicatorDuration = USERMOD_BATTERY_LOW_POWER_INDICATOR_DURATION;
+    bool lowPowerIndicationDone = false;
+    unsigned long lowPowerActivationTime = 0; // used temporary during active time
+    int8_t lastPreset = 0;
+
+    bool initDone = false;
+    bool initializing = true;
+
+    // strings to reduce flash memory usage (used more than twice)
+    static const char _name[];
+    static const char _readInterval[];
+    static const char _enabled[];
+    static const char _threshold[];
+    static const char _preset[];
+    static const char _duration[];
+    static const char _init[];
+    
+
+    // custom map function
+    // https://forum.arduino.cc/t/floating-point-using-map-function/348113/2
+    double mapf(double x, double in_min, double in_max, double out_min, double out_max) 
+    {
+      return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+    }
+
+    float truncate(float val, int8_t dec) 
+    {
+      float x = val * pow(10, dec);
+      float y = round(x);
+      float z = x - y;
+      if ((int)z == 5)
+        y++;
+      x = y / pow(10, dec);
+      return x;
+    }
+
+    /*
+     * Turn off all leds
+     */
+    void turnOff()
+    {
+      bri = 0;
+      stateUpdated(CALL_MODE_DIRECT_CHANGE);
+    }
+
+    /*
+     * Indicate low power by activating a configured preset for a given time and then switching back to the preset that was selected previously
+     */
+    void lowPowerIndicator()
+    {
+      if (!lowPowerIndicatorEnabled) return;
+      if (lowPowerIndicationDone && lowPowerIndicatorReactivationThreshold <= batteryLevel) lowPowerIndicationDone = false;
+      if (lowPowerIndicatorThreshold <= batteryLevel) return;
+      if (lowPowerIndicationDone) return;
+      if (lowPowerActivationTime <= 1) {
+        lowPowerActivationTime = millis();
+        lastPreset = currentPreset;
+        applyPreset(lowPowerIndicatorPreset);
+        colorUpdated(CALL_MODE_DIRECT_CHANGE);
+      }
+
+      if(lowPowerActivationTime+(lowPowerIndicatorDuration*1000) <= millis())
+      {
+        lowPowerIndicationDone = true;
+        lowPowerActivationTime = 0;
+        applyPreset(lastPreset);
+        colorUpdated(CALL_MODE_DIRECT_CHANGE);
+      }      
+    }
+
+  public:
+    //Functions called by WLED
+
+    /*
+     * setup() is called once at boot. WiFi is not yet connected at this point.
+     * You can use it to initialize variables, sensors or similar.
+     */
+    void setup() 
+    {
+      #ifdef ARDUINO_ARCH_ESP32
+        DEBUG_PRINTLN(F("Allocating battery pin..."));
+        if (batteryPin >= 0 && pinManager.allocatePin(batteryPin, false)) 
```

Comment:
On esp32 you need an addition condition, to check that the pin is actually supporting analogread(). 
` && (digitalPinToAnalogChannel(batterypin) >= 0)`

Reply:
Done. 👍
(see next commit)